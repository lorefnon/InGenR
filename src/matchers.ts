import * as XRegExp from "xregexp"

export interface ParseOptions {
  commentStartRegex: string
  commentEndRegex: string
  commentLBoundRegEx: string
  commentRBoundRegEx: string
}

export const getMatchers = (p: ParseOptions) => ({
  commentStartRegex: XRegExp(
    `^(?<leadingSpace> \\s*)                             # leading space before the comment block begins
      ${p.commentStartRegex}
      (?<lineBody> .*?)
      (${p.commentRBoundRegEx}|(?<commentEnd> ${p.commentEndRegex}))
      \\s*$                                                # Trailing space 
      `,
    "x"
  ),
  expandDirectiveRegex: XRegExp(
    `^\\s*InGenR:expand
      (
        \\s+                                               # Optional inline directive arguments follow:
        (?<templateName> [^,\\s]+)                            # Template name
        (?<additionalTemplateNames> (\\s*,\\s*[^,\\s]+)*)     # Additional comma separated template names
                                                              # XRegExp unfortunately doesn't allow multiple groups
                                                              # with the same name, which would have been handy here.
        (\\s+(?<configFilePath> \\S+))?                    # Name of configuration file
      )?\\s*$                                              # Trailing space 
      `,
    "x"
  ),
  directiveEndRegex: XRegExp(`^\\s*InGenR:end\\s*$`, "x"),
  blockBodyLineRegex: XRegExp(
    `^\\s*
      ((?<preCommentEnd> ${p.commentEndRegex})|(
        ${p.commentLBoundRegEx}
        (?<lineBody> .*?)
        (${p.commentRBoundRegEx}|(?<postCommentEnd> ${p.commentEndRegex}))  
      ))
      \\s*$
      `,
    "x"
  ),
  commentEndRegex: XRegExp(`${p.commentEndRegex}`)
})
