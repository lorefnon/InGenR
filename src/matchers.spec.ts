import { getMatchers } from "./matchers";
import { defaultParserOptions } from "./CommentParser";
import * as XRegExp from "xregexp";

describe("getMatchers", () => {
    const matchers = getMatchers(defaultParserOptions)
    const testSnapshot = (obj: object) => {
        expect(obj ? Object.entries(obj) : obj).toMatchSnapshot();
    }
    test("commentStartRegex", () => {
        const exec = (str: string) => XRegExp.exec(str, matchers.commentStartRegex)
        testSnapshot(exec('/*!*/'))
        testSnapshot(exec('/*! foo */'))
        testSnapshot(exec('/*! foo bar baz */'))
        testSnapshot(exec('/*! foo'))
        testSnapshot(exec('/*! foo bar *'))
        testSnapshot(exec('   /*! foo bar * '))
        testSnapshot(exec('   /*! foo bar */ '))
        testSnapshot(exec('   /*! foo bar    */ '))
    })
    test("expandDirectiveRegex", () => {
        const exec = (str: string) => XRegExp.exec(str, matchers.expandDirectiveRegex)
        testSnapshot(exec('InGenR:expand'))
        testSnapshot(exec('   InGenR:expand  '))
        testSnapshot(exec('InGenR:expand tmplName'))
        testSnapshot(exec('InGenR:expand tmpl1,templ2'))
        testSnapshot(exec('InGenR:expand tmpl1, tmpl2'))
        testSnapshot(exec('InGenR:expand tmpl1, tmpl2, tmpl3,   tmpl4,tmpl5,foo/tmpl6,bar/tmpl-7, baz/Tmpl_9'))
        testSnapshot(exec(' InGenR:expand tmpl-1 some-config.yml '))
        testSnapshot(exec('InGenR:expand tmpl1, tmpl2 some-config.yml'))
        testSnapshot(exec('InGenR:expand tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml'))
        testSnapshot(exec('  InGenR:expand tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml  '))
        testSnapshot(exec(' InGenR:expand knex-table-interface config.yml '))
    })
    test("blockBodyLineRegex", () => {
        const exec = (str: string) => XRegExp.exec(str, matchers.blockBodyLineRegex)
        testSnapshot(exec('*'))
        testSnapshot(exec('* '))
        testSnapshot(exec('* foo *'))
        testSnapshot(exec('* key: value1, value2 *'))
        testSnapshot(exec('* key: {a: value1, b: value2} *'))
        testSnapshot(exec('* key: {a: value1, b: value2}'))
        testSnapshot(exec('**/'))
        testSnapshot(exec('* */'))
        testSnapshot(exec('* foo */'))
        testSnapshot(exec('* key: value1, value2 */'))
        testSnapshot(exec('*/'))
        testSnapshot(exec(''))
    })
    test("directiveEndRegex", () => {
        const exec = (str: string) => XRegExp.exec(str, matchers.directiveEndRegex)
        testSnapshot(exec('InGenR:end'))
        testSnapshot(exec(' InGenR:end '))
    })
})