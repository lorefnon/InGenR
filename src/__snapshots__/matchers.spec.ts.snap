// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMatchers blockBodyLineRegex 1`] = `
Array [
  Array [
    "0",
    "*",
  ],
  Array [
    "1",
    "*",
  ],
  Array [
    "3",
    "*",
  ],
  Array [
    "4",
    "",
  ],
  Array [
    "5",
    "",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "*",
  ],
  Array [
    "lineBody",
    "",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 2`] = `
Array [
  Array [
    "0",
    "* ",
  ],
  Array [
    "1",
    "*",
  ],
  Array [
    "3",
    "*",
  ],
  Array [
    "4",
    "",
  ],
  Array [
    "5",
    "",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* ",
  ],
  Array [
    "lineBody",
    "",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 3`] = `
Array [
  Array [
    "0",
    "* foo *",
  ],
  Array [
    "1",
    "* foo *",
  ],
  Array [
    "3",
    "* foo *",
  ],
  Array [
    "4",
    " foo ",
  ],
  Array [
    "5",
    "*",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* foo *",
  ],
  Array [
    "lineBody",
    " foo ",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 4`] = `
Array [
  Array [
    "0",
    "* key: value1, value2 *",
  ],
  Array [
    "1",
    "* key: value1, value2 *",
  ],
  Array [
    "3",
    "* key: value1, value2 *",
  ],
  Array [
    "4",
    " key: value1, value2 ",
  ],
  Array [
    "5",
    "*",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* key: value1, value2 *",
  ],
  Array [
    "lineBody",
    " key: value1, value2 ",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 5`] = `
Array [
  Array [
    "0",
    "* key: {a: value1, b: value2} *",
  ],
  Array [
    "1",
    "* key: {a: value1, b: value2} *",
  ],
  Array [
    "3",
    "* key: {a: value1, b: value2} *",
  ],
  Array [
    "4",
    " key: {a: value1, b: value2} ",
  ],
  Array [
    "5",
    "*",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* key: {a: value1, b: value2} *",
  ],
  Array [
    "lineBody",
    " key: {a: value1, b: value2} ",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 6`] = `
Array [
  Array [
    "0",
    "* key: {a: value1, b: value2}",
  ],
  Array [
    "1",
    "* key: {a: value1, b: value2}",
  ],
  Array [
    "3",
    "* key: {a: value1, b: value2}",
  ],
  Array [
    "4",
    " key: {a: value1, b: value2}",
  ],
  Array [
    "5",
    "",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* key: {a: value1, b: value2}",
  ],
  Array [
    "lineBody",
    " key: {a: value1, b: value2}",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 7`] = `
Array [
  Array [
    "0",
    "**/",
  ],
  Array [
    "1",
    "**/",
  ],
  Array [
    "3",
    "**/",
  ],
  Array [
    "4",
    "",
  ],
  Array [
    "5",
    "*/",
  ],
  Array [
    "6",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "**/",
  ],
  Array [
    "lineBody",
    "",
  ],
  Array [
    "postCommentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 8`] = `
Array [
  Array [
    "0",
    "* */",
  ],
  Array [
    "1",
    "* */",
  ],
  Array [
    "3",
    "* */",
  ],
  Array [
    "4",
    " ",
  ],
  Array [
    "5",
    "*/",
  ],
  Array [
    "6",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* */",
  ],
  Array [
    "lineBody",
    " ",
  ],
  Array [
    "postCommentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 9`] = `
Array [
  Array [
    "0",
    "* foo */",
  ],
  Array [
    "1",
    "* foo */",
  ],
  Array [
    "3",
    "* foo */",
  ],
  Array [
    "4",
    " foo ",
  ],
  Array [
    "5",
    "*/",
  ],
  Array [
    "6",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* foo */",
  ],
  Array [
    "lineBody",
    " foo ",
  ],
  Array [
    "postCommentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 10`] = `
Array [
  Array [
    "0",
    "* key: value1, value2 */",
  ],
  Array [
    "1",
    "* key: value1, value2 */",
  ],
  Array [
    "3",
    "* key: value1, value2 */",
  ],
  Array [
    "4",
    " key: value1, value2 ",
  ],
  Array [
    "5",
    "*/",
  ],
  Array [
    "6",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "* key: value1, value2 */",
  ],
  Array [
    "lineBody",
    " key: value1, value2 ",
  ],
  Array [
    "postCommentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 11`] = `
Array [
  Array [
    "0",
    "*/",
  ],
  Array [
    "1",
    "*/",
  ],
  Array [
    "2",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "*/",
  ],
  Array [
    "preCommentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers blockBodyLineRegex 12`] = `
Array [
  Array [
    "0",
    "",
  ],
  Array [
    "1",
    "",
  ],
  Array [
    "3",
    "",
  ],
  Array [
    "4",
    "",
  ],
  Array [
    "5",
    "",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "",
  ],
  Array [
    "lineBody",
    "",
  ],
]
`;

exports[`getMatchers commentStartRegex 1`] = `
Array [
  Array [
    "0",
    "/*!*/",
  ],
  Array [
    "1",
    "",
  ],
  Array [
    "2",
    "",
  ],
  Array [
    "3",
    "*/",
  ],
  Array [
    "4",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "/*!*/",
  ],
  Array [
    "leadingSpace",
    "",
  ],
  Array [
    "lineBody",
    "",
  ],
  Array [
    "commentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers commentStartRegex 2`] = `
Array [
  Array [
    "0",
    "/*! foo */",
  ],
  Array [
    "1",
    "",
  ],
  Array [
    "2",
    " foo ",
  ],
  Array [
    "3",
    "*/",
  ],
  Array [
    "4",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "/*! foo */",
  ],
  Array [
    "leadingSpace",
    "",
  ],
  Array [
    "lineBody",
    " foo ",
  ],
  Array [
    "commentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers commentStartRegex 3`] = `
Array [
  Array [
    "0",
    "/*! foo bar baz */",
  ],
  Array [
    "1",
    "",
  ],
  Array [
    "2",
    " foo bar baz ",
  ],
  Array [
    "3",
    "*/",
  ],
  Array [
    "4",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "/*! foo bar baz */",
  ],
  Array [
    "leadingSpace",
    "",
  ],
  Array [
    "lineBody",
    " foo bar baz ",
  ],
  Array [
    "commentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers commentStartRegex 4`] = `
Array [
  Array [
    "0",
    "/*! foo",
  ],
  Array [
    "1",
    "",
  ],
  Array [
    "2",
    " foo",
  ],
  Array [
    "3",
    "",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "/*! foo",
  ],
  Array [
    "leadingSpace",
    "",
  ],
  Array [
    "lineBody",
    " foo",
  ],
]
`;

exports[`getMatchers commentStartRegex 5`] = `
Array [
  Array [
    "0",
    "/*! foo bar *",
  ],
  Array [
    "1",
    "",
  ],
  Array [
    "2",
    " foo bar ",
  ],
  Array [
    "3",
    "*",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "/*! foo bar *",
  ],
  Array [
    "leadingSpace",
    "",
  ],
  Array [
    "lineBody",
    " foo bar ",
  ],
]
`;

exports[`getMatchers commentStartRegex 6`] = `
Array [
  Array [
    "0",
    "   /*! foo bar * ",
  ],
  Array [
    "1",
    "   ",
  ],
  Array [
    "2",
    " foo bar ",
  ],
  Array [
    "3",
    "*",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "   /*! foo bar * ",
  ],
  Array [
    "leadingSpace",
    "   ",
  ],
  Array [
    "lineBody",
    " foo bar ",
  ],
]
`;

exports[`getMatchers commentStartRegex 7`] = `
Array [
  Array [
    "0",
    "   /*! foo bar */ ",
  ],
  Array [
    "1",
    "   ",
  ],
  Array [
    "2",
    " foo bar ",
  ],
  Array [
    "3",
    "*/",
  ],
  Array [
    "4",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "   /*! foo bar */ ",
  ],
  Array [
    "leadingSpace",
    "   ",
  ],
  Array [
    "lineBody",
    " foo bar ",
  ],
  Array [
    "commentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers commentStartRegex 8`] = `
Array [
  Array [
    "0",
    "   /*! foo bar    */ ",
  ],
  Array [
    "1",
    "   ",
  ],
  Array [
    "2",
    " foo bar    ",
  ],
  Array [
    "3",
    "*/",
  ],
  Array [
    "4",
    "*/",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "   /*! foo bar    */ ",
  ],
  Array [
    "leadingSpace",
    "   ",
  ],
  Array [
    "lineBody",
    " foo bar    ",
  ],
  Array [
    "commentEnd",
    "*/",
  ],
]
`;

exports[`getMatchers directiveEndRegex 1`] = `
Array [
  Array [
    "0",
    "InGenR:end",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:end",
  ],
]
`;

exports[`getMatchers directiveEndRegex 2`] = `
Array [
  Array [
    "0",
    " InGenR:end ",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    " InGenR:end ",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 1`] = `
Array [
  Array [
    "0",
    "InGenR:expand",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 2`] = `
Array [
  Array [
    "0",
    "   InGenR:expand  ",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "   InGenR:expand  ",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 3`] = `
Array [
  Array [
    "0",
    "InGenR:expand tmplName",
  ],
  Array [
    "1",
    " tmplName",
  ],
  Array [
    "2",
    "tmplName",
  ],
  Array [
    "3",
    "",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand tmplName",
  ],
  Array [
    "templateName",
    "tmplName",
  ],
  Array [
    "additionalTemplateNames",
    "",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 4`] = `
Array [
  Array [
    "0",
    "InGenR:expand tmpl1,templ2",
  ],
  Array [
    "1",
    " tmpl1,templ2",
  ],
  Array [
    "2",
    "tmpl1",
  ],
  Array [
    "3",
    ",templ2",
  ],
  Array [
    "4",
    ",templ2",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand tmpl1,templ2",
  ],
  Array [
    "templateName",
    "tmpl1",
  ],
  Array [
    "additionalTemplateNames",
    ",templ2",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 5`] = `
Array [
  Array [
    "0",
    "InGenR:expand tmpl1, tmpl2",
  ],
  Array [
    "1",
    " tmpl1, tmpl2",
  ],
  Array [
    "2",
    "tmpl1",
  ],
  Array [
    "3",
    ", tmpl2",
  ],
  Array [
    "4",
    ", tmpl2",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand tmpl1, tmpl2",
  ],
  Array [
    "templateName",
    "tmpl1",
  ],
  Array [
    "additionalTemplateNames",
    ", tmpl2",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 6`] = `
Array [
  Array [
    "0",
    "InGenR:expand tmpl1, tmpl2, tmpl3,   tmpl4,tmpl5,foo/tmpl6,bar/tmpl-7, baz/Tmpl_9",
  ],
  Array [
    "1",
    " tmpl1, tmpl2, tmpl3,   tmpl4,tmpl5,foo/tmpl6,bar/tmpl-7, baz/Tmpl_9",
  ],
  Array [
    "2",
    "tmpl1",
  ],
  Array [
    "3",
    ", tmpl2, tmpl3,   tmpl4,tmpl5,foo/tmpl6,bar/tmpl-7, baz/Tmpl_9",
  ],
  Array [
    "4",
    ", baz/Tmpl_9",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand tmpl1, tmpl2, tmpl3,   tmpl4,tmpl5,foo/tmpl6,bar/tmpl-7, baz/Tmpl_9",
  ],
  Array [
    "templateName",
    "tmpl1",
  ],
  Array [
    "additionalTemplateNames",
    ", tmpl2, tmpl3,   tmpl4,tmpl5,foo/tmpl6,bar/tmpl-7, baz/Tmpl_9",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 7`] = `
Array [
  Array [
    "0",
    " InGenR:expand tmpl-1 some-config.yml ",
  ],
  Array [
    "1",
    " tmpl-1 some-config.yml",
  ],
  Array [
    "2",
    "tmpl-1",
  ],
  Array [
    "3",
    "",
  ],
  Array [
    "5",
    " some-config.yml",
  ],
  Array [
    "6",
    "some-config.yml",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    " InGenR:expand tmpl-1 some-config.yml ",
  ],
  Array [
    "templateName",
    "tmpl-1",
  ],
  Array [
    "additionalTemplateNames",
    "",
  ],
  Array [
    "configFilePath",
    "some-config.yml",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 8`] = `
Array [
  Array [
    "0",
    "InGenR:expand tmpl1, tmpl2 some-config.yml",
  ],
  Array [
    "1",
    " tmpl1, tmpl2 some-config.yml",
  ],
  Array [
    "2",
    "tmpl1",
  ],
  Array [
    "3",
    ", tmpl2",
  ],
  Array [
    "4",
    ", tmpl2",
  ],
  Array [
    "5",
    " some-config.yml",
  ],
  Array [
    "6",
    "some-config.yml",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand tmpl1, tmpl2 some-config.yml",
  ],
  Array [
    "templateName",
    "tmpl1",
  ],
  Array [
    "additionalTemplateNames",
    ", tmpl2",
  ],
  Array [
    "configFilePath",
    "some-config.yml",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 9`] = `
Array [
  Array [
    "0",
    "InGenR:expand tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "1",
    " tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "2",
    "tmpl1",
  ],
  Array [
    "3",
    ", tmpl2",
  ],
  Array [
    "4",
    ", tmpl2",
  ],
  Array [
    "5",
    " someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "6",
    "someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "InGenR:expand tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "templateName",
    "tmpl1",
  ],
  Array [
    "additionalTemplateNames",
    ", tmpl2",
  ],
  Array [
    "configFilePath",
    "someDir/some_sub_dir/some-config.yml",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 10`] = `
Array [
  Array [
    "0",
    "  InGenR:expand tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml  ",
  ],
  Array [
    "1",
    " tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "2",
    "tmpl1",
  ],
  Array [
    "3",
    ", tmpl2",
  ],
  Array [
    "4",
    ", tmpl2",
  ],
  Array [
    "5",
    " someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "6",
    "someDir/some_sub_dir/some-config.yml",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    "  InGenR:expand tmpl1, tmpl2 someDir/some_sub_dir/some-config.yml  ",
  ],
  Array [
    "templateName",
    "tmpl1",
  ],
  Array [
    "additionalTemplateNames",
    ", tmpl2",
  ],
  Array [
    "configFilePath",
    "someDir/some_sub_dir/some-config.yml",
  ],
]
`;

exports[`getMatchers expandDirectiveRegex 11`] = `
Array [
  Array [
    "0",
    " InGenR:expand knex-table-interface config.yml ",
  ],
  Array [
    "1",
    " knex-table-interface config.yml",
  ],
  Array [
    "2",
    "knex-table-interface",
  ],
  Array [
    "3",
    "",
  ],
  Array [
    "5",
    " config.yml",
  ],
  Array [
    "6",
    "config.yml",
  ],
  Array [
    "index",
    0,
  ],
  Array [
    "input",
    " InGenR:expand knex-table-interface config.yml ",
  ],
  Array [
    "templateName",
    "knex-table-interface",
  ],
  Array [
    "additionalTemplateNames",
    "",
  ],
  Array [
    "configFilePath",
    "config.yml",
  ],
]
`;
